### ISSUE TITLE
Mysql calls are being attributed to the calling service and not treated as a remote one

### ISSUE THREAD

**Issue by rieske** [openzipkin/brave#561]
This code:
https://github.com/openzipkin/brave/blob/0b44ef88cb818a69d178f0c30e7e2ba8711b2966/instrumentation/mysql/src/main/java/brave/mysql/TracingStatementInterceptor.java#L82

expects an IP address to treat mysql call as a remote endpoint. Why?
I would like to use a domain name for my mysql connection and still have instrumentation treat it as an external service.

**Comment by adriancole** [comment]
zipkin's data format stores ip information as they can change. There's no
field for hostname, which can resolve to many different IPs. The ways to
address this include some of the following:

* use the hostname as a service name (if it is stable)
* resolve the hostname into an IP (performance hit and possibly inaccurate)
* add a tag for the hostname

We have some outstanding work to have standard tags for sql things, but
anyway let me know if anything above sounds reasonable to you

**Comment by TanyaGaleyev** [comment]
Hi,

I have the same problem. Are ipv4 and ipv6 fields of `Endpoint` required? I suspect that one of them can easily be null. If both could be null it seems that problem can be resolved by removing conditional return from `TracingStatementInterceptor` and leaving just:
```java
      builder.parseIp(connection.getHost());
      span.remoteEndpoint(builder.build());
```

**Comment by devinsba** [comment]
Yes, bot IPs are allowed to be null, since sometimes you can't resolve the IP without a DNS lookup

**Comment by TanyaGaleyev** [comment]
I expressed my suggestions in simple PR https://github.com/openzipkin/brave/pull/700

**Comment by rieske** [comment]
Since the above PR was merged in, I ported the same fix for mysql connectors 6 and 8: https://github.com/openzipkin/brave/pull/726
